pserver/gcloudstorage/storage.py

```diff
@@ -105,6 +105,14 @@ def __init__(self, context, request, field):
 
         if 'X-UPLOAD-FILENAME' in self.request.headers:
             file.filename = self.request.headers['X-UPLOAD-FILENAME']
+            try:
+                #Let clients specify a filename in base64 to preserve encoding
+                if file.filename.split()[0] == 'filename' and file.filename.split()[1] is not None:
```

Comment:
Another header is fine, I just want to be able to send up a filename with international characters.